src/cinder/gl/GlslProg.cpp

```diff
@@ -329,6 +516,57 @@ GlslProg::GlslProg( const Format &format )
 
 	link();
 	
+	cacheActiveAttribs();
+#if ! defined( CINDER_GL_ES_2 )
+	if( ! format.getVaryings().empty() ) {
+		cacheActiveTransformFeedbackVaryings();
+	}
+	cacheActiveUniformBlocks();
+#endif
+	cacheActiveUniforms();
+	
+	auto & userDefinedUniforms = format.getUniforms();
+	// check if the user thinks there's a uniform that isn't active
+	for( auto &userUniform : userDefinedUniforms ) {
+		bool foundUserDefined = false;
+		for( auto & activeUniform : mUniforms ) {
+			// if the user defined name and the active name are the same
+			if( userUniform.mName == activeUniform.mName ) {
+				foundUserDefined = true;
+				// If we've found the uniform, change the semantic to the
+				// user defined semantic.
+				activeUniform.mSemantic = userUniform.mSemantic;
+				break;
+			}
+		}
+		if( ! foundUserDefined ) {
+			CI_LOG_E( "Unknown uniform: \"" << userUniform.mName << "\"" );
+			mLoggedUniformNames.insert( userUniform.mName );
+		}
+	}
+	// make sure we get all of the semantic info correct from the user
+	for( auto &userAttrib : userDefinedAttribs ) {
+		bool active = false;
+		for( auto &activeAttrib : mAttributes ) {
+			// check if either the user defined name or user defined loc for
+			// this attrib are the same as the active attribute
+			if( userAttrib.mName == activeAttrib.mName ||
+			    userAttrib.mLoc == activeAttrib.mLoc ) {
+				// we've found the user defined attribute
+				active = true;
+				// check if the user defined attribute has cached what type
+				// of semantic this is
+				if( userAttrib.mSemantic != geom::Attrib::USER_DEFINED ) {
+					activeAttrib.mSemantic = userAttrib.mSemantic;
+				}
+				break;
+			}
+		}
+		if( !active ) {
+			CI_LOG_E( "Unknown attribute: \"" << userAttrib.mName << "\"" );
```

Comment:
Hey @ryanbartley, I hit this line because my attribute had been automatically optimized out since I wasn't using it. Would it be possible to make this message more explicit, or add a comment? Or are there more cases where this would be triggered?